From bd27b4c2b1c4e6b47988d81b1ea3b30a1cd5eb4e Mon Sep 17 00:00:00 2001
From: Jorge Zapata <jorgeluis.zapata@gmail.com>
Date: Sun, 28 Apr 2024 19:49:17 +0200
Subject: [PATCH 11/14] Initial main loop support

WebAssembly has no wakeup file descriptor to poll, so do not register
one with a new main context.

Ported: only the part that skips registering the wakeup's poll fd,
guarded by G_PLATFORM_WASM. Running g_main_loop_run() through
emscripten_set_main_loop_arg() is left out.

(cherry picked from commit 4c360061dda35cb2c5b847b4e00ec6b56dff403f)
---
 glib/gmain.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/glib/gmain.c b/glib/gmain.c
index 7c5682a18..e3c72d37e 100644
--- a/glib/gmain.c
+++ b/glib/gmain.c
@@ -711,8 +711,10 @@ g_main_context_new_with_flags (GMainContextFlags flags)
   context->time_is_fresh = FALSE;
   
   context->wakeup = g_wakeup_new ();
+#ifndef G_PLATFORM_WASM
   g_wakeup_get_pollfd (context->wakeup, &context->wake_up_rec);
   g_main_context_add_poll_unlocked (context, 0, &context->wake_up_rec);
+#endif
 
 #ifdef G_MAIN_POLL_DEBUG
   if (_g_main_poll_debug)
